### PR TITLE
created corp type module for sharing

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#6393

*Description of changes:*
- added bcrs-shared-components/corp-type-module dependency
- re-exported enums, interface and functions
- changed EntityType to CorpTypeCd
- updated other affected files
- updated unit tests
- updated app version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/interfaces/business-interface.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import { FilingTypes } from '@/enums'
+import { CorpTypeCd } from '@/enums'
 
 /** A filing's business object from the API. */
 export interface BusinessIF {
-  identifier: string;
-  legalName: string;
-  legalType?: FilingTypes;
```
Comment (severinbeauvais) [inline src/interfaces/business-interface.ts:7]: Bugfix!

File: src/mixins/enum-mixin.ts [hunk 2]
```diff
@@ -112,51 +113,24 @@ export default class EnumMixin extends Vue {
...
-      case EntityTypes.BC_ULC_COMPANY: return 'BC Unlimited Liability Company'
-      case EntityTypes.BENEFIT_COMPANY: return 'BC Benefit Company'
-      case EntityTypes.COOP: return 'BC Cooperative'
-    }
-    return '' // should never happen
-  }
+  // from external module
+  getCorpTypeInfo = GetCorpInfoObject
+  getCorpTypeDescription = GetCorpFullDescription
+  getCorpTypeNumberedDescription = GetCorpNumberedDescription
```
Comment (severinbeauvais) [inline src/mixins/enum-mixin.ts:119]: This takes the functions from the module and brings them into `this` mixin.

File: src/modules/corp-type-module.ts [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+  EXTRA_PRO_A = 'A',
+  EXTRA_PRO_B = 'B',
+  BC_COMPANY = 'BC',
+  BENEFIT_COMPANY = 'BEN',
+  CONTINUE_IN = 'C',
+  BC_CCC = 'CC',
+  CCC_CONTINUE_IN = 'CCC',
+  CEMETARY = 'CEM',
+  COOP = 'CP',
+  CORP = 'CR', // FOR NAME REQUEST ENTITY TYPE ONLY?
```
Comment (severinbeauvais) [inline src/modules/corp-type-module.ts:19]: This is carried over from previous Entity Types.

File: src/modules/corp-type-module.ts [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+  EXTRA_PRO_A = 'A',
+  EXTRA_PRO_B = 'B',
+  BC_COMPANY = 'BC',
+  BENEFIT_COMPANY = 'BEN',
+  CONTINUE_IN = 'C',
+  BC_CCC = 'CC',
+  CCC_CONTINUE_IN = 'CCC',
+  CEMETARY = 'CEM',
+  COOP = 'CP',
+  BC_CORPORATION = 'CR', // FOR NAME REQUEST ENTITY TYPE ONLY?
```
Comment (severinbeauvais) [inline src/modules/corp-type-module.ts:19]: This is carried over from Entity Types.

File: tests/unit/FilingHistoryList.spec.ts [hunk 5]
```diff
@@ -1323,7 +1323,7 @@ describe('Filing History List - Alteration Notices', () => {
             filingId: 85114
           },
           business: {
-            legalType: 'BCC' // 'from' type
```
Comment (severinbeauvais) [inline tests/unit/FilingHistoryList.spec.ts:1326]: Obsolete.
(I check the Dev and Prod dbs and no business has this legal type.)

**Review by cameron-freshworks — APPROVED** [review]
Looks great!

This is the type of pattern that would really benefit our architecture of multiple UI's. This alongside our current pattern of shared components is cool to work on! 
Certainly more work up front but can benefit long term.

**Pull request merged**